code-challenges.test.js

```diff
@@ -52,5 +110,29 @@ var numbersToAdd2 = [0, 7, -8, 12]
 var numbersToAdd3 = []
 // Expected output: []
 
+describe("sumArray", () => {
+  it("takes in an array and returns an array of the accumulating sum", () => {
+    expect(sumArray(numbersToAdd1)).toEqual([2, 6, 51, 60]);
+    expect(sumArray(numbersToAdd2)).toEqual([0, 7, -1, 11]);
+    expect(sumArray(numbersToAdd3)).toEqual([]);
+  })
+})
+
+//ReferenceError: sumArray is not defined
 
 // b) Create the function that makes the test pass.
+
+const sumArray = (array) => {
+  //declare a variable to create an empty array
+  let newArray = [];
+  //use reduce method to iterate through the array and add the value of the previous index to the value of the following index
+  array.reduce(
+    (previousValue, currentIndex, i) => (newArray[i] = previousValue + currentIndex), 0
+  );
+  return newArray;
+};
+console.log(sumArray(numbersToAdd1));
+console.log(sumArray(numbersToAdd2));
+console.log(sumArray(numbersToAdd3));
+
```

Comment:
Oooh I love reduce! Just a reminder that your test should call your function. You shouldn't need any additional function calls.